feat/add-mimir-deployment

GVengelen opened the pull request:
# Add Mimir role
This PR adds the mimir role for deploying the mimir binary.

Includes molecule tests

Since this is my first contribution I'm open to feedback!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana-ansible-collection?pullRequest=183) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana-ansible-collection?pullRequest=183) before we can accept your contribution.<br/><hr/>**Gerard** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana-ansible-collection?pullRequest=183) it.</sub>

GVengelen commented:
Hi,

I’ve added myself as a codeowner. Looking forward to contribute. I also have a Loki role which I’ll add once the Mimir role is done.

There are some Molecule tests in the role but I see that you’re using ansible-test.  I’m not familiar with this framework so I’ll try to get the gist of it.

Please let me know if there are things you would like to see different. I’m here to contribute and learn😊.

Cheers,

Gerard van Engelen
DevOps Engineer

[signature_3095144324]<https://www.openpeople.nl/>
Valueminds | Codepeople | Testpeople | Intellify
Reduitlaan 33   |   4814 DC Breda   |   The Netherlands
Office: + 31 85 273 19 54   |   Mobile: +31 6 29 42 80 60
www.openpeople.nl<http://www.openpeople.nl/>




From: Ishan Jain ***@***.***>
Date: Saturday, 13 April 2024 at 07:52
To: grafana/grafana-ansible-collection ***@***.***>
Cc: Gerard van Engelen ***@***.***>, Mention ***@***.***>
Subject: Re: [grafana/grafana-ansible-collection] feat/add-mimir-deployment (PR #183)

CC @gardar<https://github.com/gardar> @v-zhuravlev<https://github.com/v-zhuravlev>

—
Reply to this email directly, view it on GitHub<https://github.com/grafana/grafana-ansible-collection/pull/183#issuecomment-2053515595>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BHZAAINUH7JA3MYJKTNG5ITY5DBZVAVCNFSM6AAAAABGEQZVJGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDANJTGUYTKNJZGU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

GVengelen commented:
> Overall the role looks good, Just couple of things (That can be done in a seperate PR too)
> 
> * Can you add an example in the examples folder for this?
> * Do you want to do a Guest Blogpost on Grafana's blog for this role? I can ask the team if you'd like

I've added an example that uses a local s3 backend. I can write a blog as well. Added some change based on the feedback.
I see that some of the tests are failing I can look into those in a separate PR to see if I can make them work.

patcher9 commented:
Good to merge @GVengelen? LGTM from my side

GVengelen commented:
> @GVengelen I had sent you an invite for this Repo, Could you accept that before I merge this PR?

I've not re

> @GVengelen I had sent you an invite for this Repo, Could you accept that before I merge this PR?

Invitation has expired, can you try again?

GVengelen commented:
> Also re o the blog post, can you share an empty googl

https://docs.google.com/document/d/1qZOIhr71jZ2qiZQ09zdwn4jZ_gIYUe3AacKB47-xArw/edit?usp=sharing
I've also added a workflow to run the tests, will have a look at other OS in the coming weeks

GVengelen commented:
@ishanjainn, I did a rewrite based on @voidquark PR for Loki. The role is now tested for Debian, Ubuntu and RHEL.
And it follows the same pattern as the loki role which makes more sense to me.
